patterns/disposer.md

```diff
@@ -0,0 +1,84 @@
+# The Promise Disposer Pattern
```

Comment:
Its not established in JavaScript in the sense that I haven't seen another library provide it out of the box. Bluebird users are using it. 

I would say its an established pattern in other languages, since they've added special official keywords for it (C# `using` keyword, Python `with` keyword, Java `try-with-resources`). And AFAIK, they are all based on the `finally` part of `try-catch-finally` in the same way that this pattern is.

So to be honest, while its technically not "established", its so obvious and equivalent to finally-based patterns in other languages that I would vote to keep it.